QtCollider/type_codec.hpp

```diff
@@ -425,6 +438,34 @@ template <> struct TypeCodec<QVariantList>
   static void write( PyrSlot *slot, const QVariantList & );
 };
 
+template<typename QObjectT>
+struct TypeCodec<QObjectT, void>
+{
+  static QObjectT read( PyrSlot *slot )
+  {
+    return safeRead(slot);
+  }
+  
+  static QObjectT safeRead( PyrSlot *slot )
+  {
+    QObjectProxy* proxy = TypeCodec<QObjectProxy*>::safeRead(slot);
+    
+    if (proxy) {
+      QObjectT action = qobject_cast<QObjectT>(proxy->object());
+      return action;
+    } else {
+      return 0;
```

Comment:
should this be `nullptr`, and then specialize the template as

```cpp
template<typename QObjectT>
struct TypeCode<QObjectT*, void>
```

which is a bit clearer in intent and clearer as a compile error if the specialization fails?

since the casts are very explicit, this might be a nice place to use `auto`, as in

```cpp
auto proxy = TypeCode<QObjectProxy*>::safeRead(slot);
```

also, a brief comment or two on this specialization couldn't hurt :)